modules/datadict/php/datadict.class.inc

```diff
@@ -133,23 +271,223 @@ class Datadict extends \NDB_Menu_Filter
                 || strtolower(substr($key, -10)) == 'categoryid'
                 || strtolower(substr($key, -6)) == 'gender'
             ) {
-                $query .= " AND $key = '$val' ";
+                $query .= ' AND ' . $key . ' = \'' . $val . '\' ';
             } else {
                 if ($val == 'empty') {
-                    $query .= " AND COALESCE(pto.description,pt.description) = ''";
+                    $query .= ' AND COALESCE(pto.description,pt.description) = \'\'';
                 } elseif ($val=='modified') {
-                    $query .= " AND pto.name IS NOT NULL";
+                    $query .= ' AND pto.name IS NOT NULL';
                 } elseif ($val=='unmodified') {
-                    $query .= " AND pto.name IS NULL";
+                    $query .= ' AND pto.name IS NULL';
                 } else {
-                    $query .= " AND $key LIKE '$val%' ";
+                    $query .= ' AND '.$key.' LIKE \'' . $val . '%\' ';
```

Comment:
if you keep queries within " ", you don't need these extra escaped ' s. there are also escape functions in the Database class for proper escaping